bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/ObserveUtils.java

```diff
@@ -187,7 +187,14 @@ public static void startResourceObservation(Environment env, BString module, BSt
                 observerContext.setEntrypointResourceAccessor(resourceAccessor.getValue());
             }
         }
-        observerContext.setServiceName(serviceName.getValue());
+
+        String serviceNameValue = serviceName.getValue();
+        String serviceNameSuffix = System.getenv("SERVICE_NAME_SUFFIX");
```

Comment:
I thought we decided to use something like this for the env variable key. `BAL_OBSERVE_SERVICE_NAME_SUFFIX`

Reply:
@sameerajayasoma Addressed the suggestions.